diff -Nur a/Makefile b/Makefile
--- a/Makefile	2016-03-28 16:50:11.000000000 +0000
+++ b/Makefile	2016-03-28 18:39:30.861423944 +0000
@@ -380,7 +380,6 @@
 
 KBUILD_CFLAGS   := -Wall -Wundef -Wstrict-prototypes -Wno-trigraphs \
 		   -fno-strict-aliasing -fno-common \
-		   -Werror-implicit-function-declaration \
 		   -Wno-format-security \
 		   -fno-delete-null-pointer-checks
 KBUILD_AFLAGS_KERNEL :=
